### ISSUE TITLE
post processing of whisperbank is missing

### ISSUE THREAD

**Issue by jie-chen** [csukuangfj/kaldi-native-fbank#63]
in https://github.com/csukuangfj/kaldi-native-fbank/blob/master/kaldi-native-fbank/csrc/whisper-feature.cc#L166, it seems after the computation of melbank, some post processings are missing from https://github.com/csukuangfj/kaldifeat/blob/master/kaldifeat/csrc/whisper-fbank.cc#L81

**Comment by csukuangfj** [comment]
You should do it by yourself, see

https://github.com/k2-fsa/sherpa-onnx/blob/e3280027f95cff53b21588d721404009695ffda2/sherpa-onnx/csrc/offline-whisper-model.cc#L438

```cpp
void OfflineWhisperModel::NormalizeFeatures(float *features, int32_t num_frames,
                                            int32_t feat_dim) {
  // log_spec = torch.clamp(features, min=1e-10).log10()
  // log_spec = torch.maximum(log_spec, log_spec.max() - 8.0)
  // mel = (log_spec + 4.0) / 4.0

  int32_t n = num_frames * feat_dim;
  float max_v = -1e20;
  for (int32_t i = 0; i != n; ++i) {
    float f = features[i];

    f = std::max<float>(f, 1e-10);
    f = std::log10(f);

    max_v = std::max(f, max_v);

    features[i] = f;
  }

  max_v -= 8;

  for (int32_t i = 0; i != n; ++i) {
    float f = features[i];
    f = std::max(f, max_v);

    f = (f + 4) / 4;

    features[i] = f;
  }
}

```

**Comment by jie-chen** [comment]
Hi, after postprocessing, the output is still not same as whisper.log_mel_spectrogram, below is a unit test


`#!/usr/bin/env python3

import kaldi_native_fbank as knf
import torch
import whisper


def main():
    sampling_rate = 16000
    samples = torch.randn(16000 * 10)
    opts = knf.WhisperFeatureOptions()

    opts.dim = 80
    online_whisper_fbank = knf.OnlineWhisperFbank(opts)

    online_whisper_fbank.accept_waveform(sampling_rate=sampling_rate, waveform=samples.numpy())
    online_whisper_fbank.input_finished()

    features = []
    for i in range(online_whisper_fbank.num_frames_ready):
        f = online_whisper_fbank.get_frame(i)
        f = torch.from_numpy(f)
        features.append(f)

    features = torch.stack(features).T
    log_spec = torch.clamp(features, min=1e-10).log10()
    log_spec = torch.maximum(log_spec, log_spec.max() - 8.0)
    mel = (log_spec + 4.0) / 4.0
    whisper_mel = whisper.log_mel_spectrogram(samples)
    for i in range(online_whisper_fbank.num_frames_ready):
        f1 = mel[:, i]
        f2 = whisper_mel[:, i]
        assert torch.allclose(f1, f2, atol=1e-3), (i, (f1 - f2).abs().max())



if __name__ == "__main__":
    torch.manual_seed(20220825)
    main()
`
